File: programmers/hash/camouflage.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map< string, vector< string > >  result;
    for (auto vs : clothes) {
        result[vs[1]].push_back(vs[0]);
    }
    for (auto vs : result) {
        answer *= (vs.second.size() + 1);
    }
    return answer - 1;
}

File: programmers/hash/camouflage.py
def solution(clothes):
    answer = 1
    result = {}
    for value, key in clothes:
        result[key] = result[key] + 1 if key in result else 1
    for key, value in result.items():
        answer *= (value + 1)
    return answer - 1

if __name__ == "__main__":
    clothes = [["yellow_hat", "headgear"]]
    solution(clothes)